onnxruntime/python/tools/transformers/fusion_attention.py

```diff
@@ -693,12 +692,15 @@ def create_multihead_attention_node(
             mha_inputs.append("")
 
         # Add optional inputs for MHA
-        if past_k and past_v and past_k in graph_input_names and past_v in graph_input_names:
+
+        if past_k and past_v:
             mha_inputs.extend([key_padding_mask, add_qk, past_k, past_v])
+        elif key_padding_mask or add_qk:
+            mha_inputs.extend([key_padding_mask, add_qk])
 
         # Add outputs for MHA
         mha_outputs = [output]
-        if present_k and present_v and present_k in graph_output_names and present_v in graph_output_names:
+        if present_k and present_v:
```

Comment:
Since the Whisper test cases are failing in the CI pipelines, the checks that `present_k` and `present_v` are in `graph_output_names` may be needed

Reply:
They are failing in the CI pipeline because it is not able to find the file mentioned. graph_output_names is required because we are anyways appending present_k and present_v even though it is empty right?

CI pipeline is failing because of the above naming issue I think. Because locally these tests are passing, and earlier when conformer tests were failing these tests were passing so the issue with the cmake changes I made

Reply:
Tests passed